errors.go

```diff
@@ -22,15 +22,51 @@ package secboot
 import (
 	"errors"
 	"fmt"
+
+	"github.com/chrisccoulson/go-tpm2"
 )
 
 var (
+	// ErrTPMClearRequiresPPI is returned from ProvisionTPM and indicates that clearing the TPM must be performed via
+	// the Physical Presence Interface.
+	ErrTPMClearRequiresPPI = errors.New("clearing the TPM requires the use of the Physical Presence Interface")
+
 	// ErrTPMProvisioning indicates that the TPM is not provisioned correctly for the requested operation. Please note that other errors
 	// that can be returned may also be caused by incomplete provisioning, as it is not always possible to detect incomplete or
 	// incorrect provisioning in all contexts.
 	ErrTPMProvisioning = errors.New("the TPM is not correctly provisioned")
+
+	// ErrTPMLockout is returned from any function when the TPM is in dictionary-attack lockout mode. Until
+	// the TPM exits lockout mode, the key will need to be recovered via a mechanism that is independent of
+	// the TPM (eg, a recovery key)
+	ErrTPMLockout = errors.New("the TPM is in DA lockout mode")
 )
 
+
+// TPMResourceExistsError is returned from any function that creates a persistent TPM resource if a resource already exists
+// at the specified handle.
+type TPMResourceExistsError struct {
+	Handle tpm2.Handle
+}
+
+func (e TPMResourceExistsError) Error() string {
+	return fmt.Sprintf("a resource already exists on the TPM at handle 0x%08x", e.Handle)
+}
+
+// AuthFailError is returned when an authorization check fails. The provided handle indicates the resource for which authorization
+// failed. Whilst the error normally indicates that the provided authorization value is incorrect, it may also be returned
+// for other reasons that would cause a HMAC check failure, such as a communication failure between the host CPU and the TPM
+// or the name of a resource on the TPM not matching the name of the ResourceContext passed to the function that failed - this
+// latter issue can occur when using a resource manager if another process accesses the TPM and makes changes to persistent
+// resources or sessions.
+type AuthFailError struct {
+	Handle tpm2.Handle
+}
+
+func (e AuthFailError) Error() string {
+	return fmt.Sprintf("an authorization check failed for the hierarchy associated with %v", e.Handle)
```

Comment:
That could possibly work, but I should actually check that "hierarchy" is appropriate here. That was the case when the error was first added, but it might not be now.

Reply:
Thinking about this - "cannot access hierarchy at handle %v" probably isn't specific enough - there are other reasons a hierarchy can be inaccessible other than an authorization failure (eg, they can be disabled by the owner). How about "cannot access resource at handle %v because an authorization check failed"?

Reply:
the longer form is fine, and yes would be better to start with "cannot..."